pkg/bmc/client/bmclib/bmclib.go

```diff
@@ -0,0 +1,100 @@
+package bmclib
```

Comment:
Can we call this package something else? This is easily confused with `bmc-toolbox/bmclib`

Reply:
Avoid creating taxonomy with packages - every directory is a package and _should_ provide something `/pkg/bmclib` is fine.

Reply:
That said, I feel like this package has functions for constructing a bmclib.Client only. If the other changes are enacted (invariant comments and removal of `== nil` checks) then all the methods become redundant and the code greatly simplified.

Reply:
> Avoid creating taxonomy with packages - every directory is a package and should provide something /pkg/bmclib is fine.

Or something other than `bmclib` to mitigate Micah's concern. I don't think I read his comment properly when I wrote my response 🤦🏻